examples/linear_model/plot_tweedie_regression_insurance_claims.py

```diff
@@ -0,0 +1,522 @@
+#          Roman Yurchak <rth.yurchak@gmail.com>
+# License: BSD 3 clause
+from functools import partial
+
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+
+from sklearn.datasets import fetch_openml
+from sklearn.compose import ColumnTransformer
+from sklearn.linear_model import PoissonRegressor, GammaRegressor
+from sklearn.linear_model import TweedieRegressor
+from sklearn.model_selection import train_test_split
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import FunctionTransformer, OneHotEncoder
+from sklearn.preprocessing import StandardScaler, KBinsDiscretizer
+
+from sklearn.metrics import mean_absolute_error, mean_squared_error
+
+
+def load_mtpl2(n_samples=100000):
+    """Fetch the French Motor Third-Party Liability Claims dataset.
+
+    Parameters
+    ----------
+    n_samples: int, default=100000
+      number of samples to select (for faster run time).
+    """
+
+    # freMTPL2freq dataset from https://www.openml.org/d/41214
+    df_freq = fetch_openml(data_id=41214, as_frame=True)['data']
+    df_freq['IDpol'] = df_freq['IDpol'].astype(np.int)
+    df_freq.set_index('IDpol', inplace=True)
+
+    # freMTPL2sev dataset from https://www.openml.org/d/41215
+    df_sev = fetch_openml(data_id=41215, as_frame=True)['data']
+
+    # sum ClaimAmount over identical IDs
+    df_sev = df_sev.groupby('IDpol').sum()
+
+    df = df_freq.join(df_sev, how="left")
+    df["ClaimAmount"].fillna(0, inplace=True)
+
+    # unquote string fields
+    for column_name in df.columns[df.dtypes.values == np.object]:
+        df[column_name] = df[column_name].str.strip("'")
+    return df.iloc[:n_samples]
+
+
+def plot_obs_pred(df, feature, weight, observed, predicted, y_label=None,
+                  title=None, ax=None):
+    """Plot observed and predicted - aggregated per feature level.
+
+    Parameters
+    ----------
+    df : DataFrame with at least three columns named feature, weight and
+         observed
+    feature: str
+        a column name of df for the feature to be plotted
+    weight : str
+        column name of df with the values of weights or exposure
+    observed : str
+        a column name of df with the observed target
+    predicted : frame
+        a dataframe, with the same index as df, with the predicted target
+    """
+    # aggregate observed and predicted variables by feature level
+    df_ = df.loc[:, [feature, weight]].copy()
+    df_["observed"] = df[observed] * df[weight]
+    df_["predicted"] = predicted * df[weight]
+    df_ = (
+        df_.groupby([feature])[weight, "observed", "predicted"]
+        .sum()
+        .assign(observed=lambda x: x["observed"] / x[weight])
+        .assign(predicted=lambda x: x["predicted"] / x[weight])
+    )
+
+    ax = df_.loc[:, ["observed", "predicted"]].plot(style=".", ax=ax)
+    y_max = df_.loc[:, ["observed", "predicted"]].values.max() * 0.8
+    ax.fill_between(
+        df_.index,
```

Comment:
> I don't understand what the plot is showing and the code and explanation don't really help.

This example has been significantly reworked since august so I'm going to mark it as resolved. Please unresolve if still unclear.